kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -331,6 +331,8 @@ private void commitOnTimeIfNoData(long now) {
         setNextScheduledRotation();
 
         commitFiles();
+      } else if (recordCount == 0 && shouldApplyScheduledRotation(now)) {
```

Comment:
Do you need to call shouldApplyScheduledRotation(now) here?  Why not always call setNextScheduledRotation() for setNextScheduledRotation?  This might be just about to expire when the first one comes in, and then the same thing will happen -- split after too few records?

Reply:
We need to check against now because some people might have the setting as 1 hour and they will complain I sent records at 11:59 but they are appearing after an hour.

Reply:
Also put with empty records list could be called 100s of time within a ScheduledRotation so we do not want to reset it on every call.

Reply:
I don;t have a strong opinion about it, but I am a little confused.  Why wouldn't you want setNextScheduledRotation() called every time there's no records received yet?  It's a relatively inexpensive call.  I was thinking that since these schedules are relative times (durations), they'd want the clock to start ticking at the first record written so that each file is written to for the same amount of time.